build-plugins/src/main/kotlin/aws/sdk/kotlin/gradle/codegen/dsl/SmithyKotlinPluginSettings.kt

```diff
@@ -0,0 +1,170 @@
+}
+
+class SmithyKotlinPluginSettings : SmithyBuildPluginSettings {
+    override val pluginName: String = "kotlin-codegen"
+
+    var serviceShapeId: String? = null
+    var packageName: String? = null
+    var packageVersion: String? = null
+    var packageDescription: String? = null
+    var sdkId: String? = null
+
+    internal var buildSettings: SmithyKotlinBuildSettings? = null
+    fun buildSettings(configure: SmithyKotlinBuildSettings.() -> Unit) {
+        if (buildSettings == null) buildSettings = SmithyKotlinBuildSettings()
+        buildSettings!!.apply(configure)
+    }
+
+    internal var apiSettings: SmithyKotlinApiSettings? = null
+    fun apiSettings(configure: SmithyKotlinApiSettings.() -> Unit) {
+        if (apiSettings == null) apiSettings = SmithyKotlinApiSettings()
+        apiSettings!!.apply(configure)
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as SmithyKotlinPluginSettings
+
+        if (serviceShapeId != other.serviceShapeId) return false
+        if (packageName != other.packageName) return false
+        if (packageVersion != other.packageVersion) return false
+        if (packageDescription != other.packageDescription) return false
+        if (sdkId != other.sdkId) return false
+        if (buildSettings != other.buildSettings) return false
+        if (apiSettings != other.apiSettings) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = serviceShapeId?.hashCode() ?: 0
+        result = 31 * result + (packageName?.hashCode() ?: 0)
+        result = 31 * result + (packageVersion?.hashCode() ?: 0)
+        result = 31 * result + (packageDescription?.hashCode() ?: 0)
+        result = 31 * result + (sdkId?.hashCode() ?: 0)
+        result = 31 * result + (buildSettings?.hashCode() ?: 0)
+        result = 31 * result + (apiSettings?.hashCode() ?: 0)
+        return result
+    }
+
+    override fun toNode(): Node {
+        val obj = ObjectNode.objectNodeBuilder()
+            .withMember("service", serviceShapeId!!)
+            .withObjectMember("package") {
+                withMember("name", packageName!!)
+                withNullableMember("version", packageVersion)
+                withNullableMember("description", packageDescription)
+            }
+            .withNullableMember("sdkId", sdkId)
+            .withNullableMember("build", buildSettings)
+            .withNullableMember("api", apiSettings)
+
+        return obj.build()
+    }
+
+    override fun toString(): String =
+        "SmithyKotlinPluginSettings(pluginName='$pluginName', serviceShapeId=$serviceShapeId, packageName=$packageName, packageVersion=$packageVersion, packageDescription=$packageDescription, sdkId=$sdkId, buildSettings=$buildSettings, apiSettings=$apiSettings)"
```

Comment:
question: is there any reason there's quotes around `'pluginName'` but nothing else?

Reply:
Auto generated by Intellij 🤷‍♂️